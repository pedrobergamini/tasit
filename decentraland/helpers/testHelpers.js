import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () =>
    "0x1234567890123456789012345678901234567890123456789012345678901234",
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
};
