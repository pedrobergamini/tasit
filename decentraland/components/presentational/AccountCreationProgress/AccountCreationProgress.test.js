import React from "react";
import { shallow } from "enzyme";
import {
  AccountCreationProgress,
  ProgressMessageAndLink,
} from "./AccountCreationProgress";
// Not the @presentational version since it needs
// the internal component
import status from "@constants/AccountCreationStatus";
const {
  NOT_STARTED,
  GENERATING_ACCOUNT,
  FUNDING_WITH_ETH,
  FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE,
  FUNDING_WITH_MANA,
  APPROVING_MARKETPLACE,
  READY_TO_USE,
} = status;
import { anAction } from "@helpers/testHelpers";

describe("AccountCreationProgress", () => {
  it("initial state - before start account setup", async () => {
    const status = NOT_STARTED;
    const actions = {};

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for account generation", async () => {
    const status = GENERATING_ACCOUNT;
    const actions = {};

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for ETH funding", async () => {
    const status = FUNDING_WITH_ETH;
    const actionId = await anAction.getId();
    const actions = { FUNDING_WITH_ETH: actionId };

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for MANA funding and marketplace approval", async () => {
    const status = FUNDING_WITH_MANA_AND_APPROVING_MARKETPLACE;
    const actionId = await anAction.getId();
    const actions = {
      FUNDING_WITH_MANA: actionId,
      APPROVING_MARKETPLACE: actionId,
    };

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for MANA funding", async () => {
    const status = FUNDING_WITH_MANA;
    const actionId = await anAction.getId();
    const actions = { FUNDING_WITH_MANA: actionId };

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("waiting for setup - waiting for marketplace approval", async () => {
    const status = APPROVING_MARKETPLACE;
    const actionId = await anAction.getId();
    const actions = { APPROVING_MARKETPLACE: actionId };

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  it("final state - account ready", async () => {
    const status = READY_TO_USE;
    const actions = {};

    expect(
      shallow(<AccountCreationProgress status={status} actions={actions} />)
    ).toMatchSnapshot();
  });

  describe("ProgressMessageAndLink", () => {
    it("with a message and without an action", async () => {
      const waitingMessage = "A waiting message.";
      const actionId = null;

      expect(
        shallow(
          <ProgressMessageAndLink
            waitingMessage={waitingMessage}
            actionId={actionId}
          />
        )
      ).toMatchSnapshot();
    });

    it("with a message and with an action", async () => {
      const waitingMessage = "A waiting message.";
      const actionId = await anAction.getId();

      expect(
        shallow(
          <ProgressMessageAndLink
            waitingMessage={waitingMessage}
            actionId={actionId}
          />
        )
      ).toMatchSnapshot();
    });

    it("without a message nor action", async () => {
      const waitingMessage = null;
      const actionId = null;

      expect(
        shallow(
          <ProgressMessageAndLink
            waitingMessage={waitingMessage}
            actionId={actionId}
          />
        )
      ).toMatchSnapshot();
    });
  });
});
