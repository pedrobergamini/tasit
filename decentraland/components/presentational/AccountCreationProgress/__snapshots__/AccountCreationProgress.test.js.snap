// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCreationProgress ProgressMessageAndLink with a message and with an action 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "maxWidth": 600,
      }
    }
  >
    <Component
      adjustsFontSizeToFit={true}
      numberOfLines={2}
      style={
        Object {
          "color": "#CCCCCC",
        }
      }
    >
      A waiting message.
    </Component>
  </Component>
  <Component>
    <LinkToBlockchain
      actionId="0x1234567890123456789012345678901234567890"
    />
  </Component>
</Component>
`;

exports[`AccountCreationProgress ProgressMessageAndLink with a message and without an action 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "maxWidth": 600,
      }
    }
  >
    <Component
      adjustsFontSizeToFit={true}
      numberOfLines={2}
      style={
        Object {
          "color": "#CCCCCC",
        }
      }
    >
      A waiting message.
    </Component>
  </Component>
  <Component>
    <LinkToBlockchain
      actionId={null}
    />
  </Component>
</Component>
`;

exports[`AccountCreationProgress ProgressMessageAndLink without a message nor action 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "maxWidth": 600,
      }
    }
  >
    <Component
      adjustsFontSizeToFit={true}
      numberOfLines={2}
      style={
        Object {
          "color": "#CCCCCC",
        }
      }
    />
  </Component>
  <Component>
    <LinkToBlockchain
      actionId={null}
    />
  </Component>
</Component>
`;

exports[`AccountCreationProgress final state - account ready 1`] = `""`;

exports[`AccountCreationProgress initial state - before start account setup 1`] = `""`;

exports[`AccountCreationProgress waiting for setup - waiting for ETH funding 1`] = `
<ProgressMessageAndLink
  actionId="0x1234567890123456789012345678901234567890"
  key="FUNDING_WITH_ETH"
  waitingMessage="Funding account with ETH..."
/>
`;

exports[`AccountCreationProgress waiting for setup - waiting for MANA funding 1`] = `
<ProgressMessageAndLink
  actionId="0x1234567890123456789012345678901234567890"
  key="FUNDING_WITH_MANA"
  waitingMessage="Funding account with MANA..."
/>
`;

exports[`AccountCreationProgress waiting for setup - waiting for MANA funding and marketplace approval 1`] = `
Array [
  <ProgressMessageAndLink
    actionId="0x1234567890123456789012345678901234567890"
    key="FUNDING_WITH_MANA"
    waitingMessage="Funding account with MANA..."
  />,
  <ProgressMessageAndLink
    actionId="0x1234567890123456789012345678901234567890"
    key="APPROVING_MARKETPLACE"
    waitingMessage="Approving marketplace..."
  />,
]
`;

exports[`AccountCreationProgress waiting for setup - waiting for account generation 1`] = `
<ProgressMessageAndLink
  key="GENERATING_ACCOUNT"
  waitingMessage="Generating account..."
/>
`;

exports[`AccountCreationProgress waiting for setup - waiting for marketplace approval 1`] = `
<ProgressMessageAndLink
  actionId="0x1234567890123456789012345678901234567890"
  key="APPROVING_MARKETPLACE"
  waitingMessage="Approving marketplace..."
/>
`;
